src/model/deepdives.py

```diff
@@ -48,28 +59,71 @@ class Warning(str, Enum):
     SHIELD_DISRUPTION = "Shield Disruption"
     SWARMAGEDDON = "Swarmageddon"
 
+class Mission():
+    """ Represents one mission objective like collect 7 Aquarq or mine 200 Morkite"""
+    name: str
+
+    def __init__(self, name: str):
+        self.name = name
+
+    @property
+    def emoji(self) -> str:
+        return get_emoji(self.type)
+
+    @property
+    def type(self) -> MissionType:
+        """Mission types provided by API don't have much overlap with regular
+        mission types thus, there needs to be such conversion"""
+
+        type_name_matrix: dict[MissionType] = {
+            'morkite': MissionType.MINING_EXPEDITION,
+            'egg':  MissionType.EGG_HUNT,
+            'refining': MissionType.ON_SITE_REFINING,
+            'mule': MissionType.SALVAGE_OPERATION,
+            'aquarq': MissionType.POINT_EXTRACTION,
+            'escort':  MissionType.ESCORT_DUTY,
+            'dreadnought': MissionType.ELIMINATION,
+            'black box':  MissionType.BLACK_BOX,
+            'sabotage':  MissionType.INDUSTRIAL_SABOTAGE
+        }
+
+        for (name, type) in type_name_matrix.items():
+            if name in self.name.lower():
+                return type
+
+        raise ValueError(f'Invalid mission name: {self.name}')
```

Comment:
Now thinking about it it might be good idea to move that to https://github.com/MoritzHayden/drg-api in future and replace code here with just data from the API

Reply:
It could be moved to the API, but that would require a contract change and consequent version bump for the API. You're more than welcome to undertake the effort, but it probably isn't required for such a small change